stellar-auth/pkg/auth/authenticator.go

```diff
@@ -477,6 +478,45 @@ func (a *defaultAuthenticator) GetUser(ctx context.Context, userID string) (*Use
 	}, nil
 }
 
+// GetUsers retrieves the respective users from a list of user IDs.
+func (a *defaultAuthenticator) GetUsers(ctx context.Context, userIDs []string) ([]*User, error) {
+	const query = `
+		SELECT
+			id,
+			first_name,
+			last_name
+		FROM
+			auth_users
+		WHERE
+			id = ANY($1::text[]) AND is_active = true
+	`
+
+	var dbUsers []authUser
+	err := a.dbConnectionPool.SelectContext(ctx, &dbUsers, query, pq.Array(userIDs))
+	if err != nil {
+		return nil, fmt.Errorf("error querying user IDs: %w", err)
+	}
+	if len(dbUsers) != len(userIDs) {
+		return nil,
+			fmt.Errorf(
+				"error querying user IDs: searching for %d users, found %d users",
+				len(userIDs),
+				len(dbUsers),
+			)
+	}
+
+	users := make([]*User, len(dbUsers))
+	for i, u := range dbUsers {
+		users[i] = &User{
+			ID:        u.ID,
+			FirstName: u.FirstName,
+			LastName:  u.LastName,
+		}
+	}
```

Comment:
I decided to keep this to preserve the separation of concerns between the 2 types of structs even if it means having to loop over and retrieve the same values from the result.